### PR TITLE
feat: endpoint for returning resource accessors

### PR DESCRIPTION
**Pull request by bmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-333

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java [hunk 1]
```diff
@@ -194,4 +203,37 @@ public void testFilterResourceWithSecurityClassification() throws Exception {
...
+        Mockito.when(filterResourceService.returnResourceAccessors(resourceId, resourceName, resourceType))
+            .thenReturn(resourceAccessorsEnvelope);
+
+        this.mockMvc.perform(get("/api/resource/resourceType/" + resourceType + "/resourceName/"
+            + resourceName + "/resourceId/" + resourceId))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.resourceId", is("0011")))
+            .andExpect(jsonPath("$.explicitAccess").exists())
+            .andExpect(jsonPath("$.explicitAccess[0].accessorId", is("5511")))
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: there should be a test case for HTTP Status code - 404 when the search resource is not found

**Review by shashank-rastogi — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java [hunk 2]
```diff
@@ -85,4 +88,20 @@ public FilteredResourceEnvelope filterResource(@RequestBody FilterResource reque
...
+    @ApiOperation(value = "Finds users who have explicit access to a resource",
+        response = ResourceAccessorsEnvelope.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successfully returns users who have access to a resource"),
+        @ApiResponse(code = 401, message = "You are not authorized to perform this particular request. "
+            + "Please provide a valid access token in the request"),
+        @ApiResponse(code = 400, message = "Incomplete request information or malformed input request"),
+        @ApiResponse(code = 500, message = "Internal server error")
+    })
+    @GetMapping("/resource/resourceType/{resourceType}/resourceName/{resourceName}/resourceId/{resourceId}")
```
[the remark this answers is not in the document]
Comment (bmcts) [reply src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java]: 👍  returns empty array as discussed

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java (hunk 1 above)
Comment (bmcts) [reply src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: 👍  returns empty array as discussed

Comment (shashank-rastogi) [reply src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: Ben - there should be an API method for this test case as this is important scenario for a GET API request

Comment (shashank-rastogi) [reply src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: oh i guess we discussed this and agreed to add this to Serenity tests once they are ready!

**Review by shashank-rastogi — APPROVED** [review]

**Pull request merged**